packages/synthetix-main/contracts/storage/MarketManagerStorage.sol

```diff
@@ -25,6 +25,8 @@ contract MarketManagerStorage {
         // used to attach/reattach pools to a market if it goes below a certain debt per debt share
         HeapUtil.Data outRangePools;
         SharesLibrary.Distribution debtDist;
+        // @notice the amount of a given collateral type deposited by this market
+        mapping(address => uint128) depositedCollateral;
```

Comment:
as mentioned up above, it might be better to store this as an `DepositCollateralEntry[] depositedCollateral` and iterate over the list rather than creating a mapping which we have to scan all collateral types for